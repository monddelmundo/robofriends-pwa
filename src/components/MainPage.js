import React, { useEffect, useState } from "react";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import ErrorBoundry from "../components/ErrorBoundry";
import "./MainPage.css";
import Header from "../components/Header";

function MainPage(props) {
  const [filteredRobots, setFilteredRobots] = useState([]);
  const {
    searchField,
    onSearchChange,
    robots,
    onRequestRobots,
    isPending,
  } = props;

  //componentDidMount
  useEffect(() => {
    componentDidMount();
  }, [robots.length]); //notice the empty array here

  async function componentDidMount() {
    if (robots.length === 0) onRequestRobots();
    if (robots.length > 0) setFilteredRobots(robots);
  }

  function filterRobots(robots) {
    return robots.filter((robot) =>
      robot.name.toLowerCase().includes(searchField.toLowerCase())
    );
  }

  useEffect(() => {
    if (robots.length) {
      setFilteredRobots(filterRobots(robots));
    }
  }, [searchField.length]);

  //tc = text center
  return isPending ? (
    <h1 name="loading">Loading...</h1>
  ) : (
    <div className="tc">
      <Header />
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filteredRobots} />
        </ErrorBoundry>
      </Scroll>
    </div>
  );
}

export default MainPage;
