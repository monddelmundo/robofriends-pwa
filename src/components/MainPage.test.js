import React from "react";
import MainPage from "./MainPage";
import renderer from "react-test-renderer";

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };

  wrapper = renderer.create(<MainPage {...mockProps} />);
});

it("renders mainpage without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});
