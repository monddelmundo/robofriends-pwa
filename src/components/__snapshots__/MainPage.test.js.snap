// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mainpage without crashing 1`] = `
<div
  className="tc"
>
  <h1
    className="f2"
  >
    Robofriends
  </h1>
  <div
    className="pa2"
  >
    <input
      aria-label="Search Robots"
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="Search Robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "600px",
        "overflowY": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
